vhdl_lang/src/analysis/tests/resolves_names.rs

```diff
@@ -1046,3 +1046,33 @@ end architecture;
 ",
     );
 }
+
+#[test]
+fn block_names_are_visible() {
+    check_code_with_no_diagnostics(
+        "
+entity ent is
+  port (ent_in : integer);
+end entity;
+
+architecture a of ent is
+  signal sig : integer;
+begin
+  blk: block (ent_in = 1) is
+    generic( gen : integer := 0 );
+    generic map ( gen => 1);
+    port( 
+      prt_in : in integer := 0;
+      prt_out : out integer := 0
+    );
+    port map (
+      prt_in => ent_in + sig,
+      prt_out => open
+    );
+  begin
+    prt_out <= gen + prt_in + sig;
```

Comment:
Are these types of blocks allowed to see the outside signals such as `sig` in this case?

Reply:
LRM 11.2 states 

> If a block header appears in a block statement, it explicitly identifies certain values or signals that are to be imported from the enclosing environment into the block and associated with formal generics or ports.

Which I would interpret as `sig` not being visible but it's not 100% clear. ModelSim will compile the snippet without errors.

Reply:
What does GHDL do? I suspect this corner of the standard is very rarely visited so there might be bugs.

Reply:
I tried it with GHDL and some other simulators/editors as well, in all of them `sig` is visible. To me it makes more sense that adding a block header would hide all objects external to the block, both from interpreting the standard and in a general sense. The VITAL standard requires models to be structured into a set of blocks and I think that these blocks must use generics and ports.   
I would be fine with allowing `sig` to be visible as that seems to be inline with most other tools as a first step. Not sure how much work it is to hide the outer scope.

Reply:
Yes to me it would also seem to make more sense to hide the enclosing scope when using generics. Then a block with generics would be equivalent with an inlined entity instantiation. I do not think this code is very common though.